src/query/pools/query_pools.go

```diff
@@ -69,20 +69,20 @@ func BuildWorkerPools(
 	instrumentOptions instrument.Options,
 	readPoolPolicy, writePoolPolicy xconfig.WorkerPoolPolicy,
 	scope tally.Scope,
-) (xsync.PooledWorkerPool, xsync.PooledWorkerPool, error) {
-	opts, readPoolSize := readPoolPolicy.Options()
+) (xsync.StaticPooledWorkerPool, xsync.DynamicPooledWorkerPool, error) {
```

Comment:
Were they both really using different pool types?

Reply:
well it depends on how you had them configured in yaml, which potentially makes this changes backwards incompatible. a user might have configured the write pool to be static.

since there was no default for read pool (like the write pool), it defaults to the zero values of the yaml config, which is grow=false.